tools/catalog_builder/catalog_builder.py

```diff
@@ -0,0 +1,326 @@
+# the directory where the output will be written, the CASENAME to use for the output file names,
+# the file names that will be linked,
+# and the frequencies and variable names to use in the new symlinked file names
+# Output: esm-intake catalog in csv format with the file locations and metadata required
+# for the preprocessor and the MDTF-diagnostics framework
+
+import click
+import intake
+import os
+import pathlib
+import sys
+import time
+import traceback
+import typing
+import xarray as xr
+import yaml
+from datetime import datetime, timedelta
+from ecgtools import Builder
+from ecgtools.builder import INVALID_ASSET, TRACEBACK
+from ecgtools.parsers import parse_cmip6
+from ecgtools.parsers.cesm import parse_cesm_timeseries
+
+
+# The ClassMaker is cribbed from SO
+# https://stackoverflow.com/questions/1176136/convert-string-to-python-class-object
+# Classmaker and the @catalog_class.maker decorator allow class instantiation from
+# strings. The main block can simply call the desired class using the convention
+# argument instead of messy if/then/else blocks. Yes, both work, but I wanted
+# to try something that, if it is not more "Pythonic", is more extensible
+
+class ClassMaker:
+    def __init__(self):
+        self.classes = {}
+
+    def add_class(self, c):
+        self.classes[c.__name__] = c
+
+    # define the class decorator to return the class passed
+    def maker(self, c):
+        self.add_class(c)
+        return c
+
+    def __getitem__(self, n):
+        return self.classes[n]
+
+
+# instantiate the class maker
+catalog_class = ClassMaker()
+
+
+# custom parser for pp data stored on GFDL archive filesystem
+def parse_gfdl_pp_ts(file_name: str):
+    # files = sorted(glob.glob(os.path.join(file_name,'*.nc')))  # debug comment when ready to run
+    # file = pathlib.Path(files[0])  # debug comment when ready to run
+    file = pathlib.Path(file_name)  # uncomment when ready to run
+
+    try:
+        # isolate file from rest of path
+        stem = file.stem
+        # split the file name into components based on _
+        split = stem.split('.')
+        realm = split[0]
+        time_range = split[1]
+        variable_id = split[2]
+        source_type = file.parts[3]
+        member_id = file.parts[4]
+        experiment_id = file.parts[5]
+        source_id = file.parts[6]
+        freq = file.parts[10]
+        chunk_freq = file.parts[11]
+        variant_label = ""
+        grid_label = ""
+        table_id = ""
+        if 'mon' in freq.lower():
+            output_frequency = 'mon'
+        elif 'day' in freq.lower():
+            output_frequency = 'day'
+        elif '6hr' in freq.lower():
+            output_frequency = '6hr'
+        elif 'subhr' in freq.lower():
+            output_frequency = 'subhr'
+
+        # call to xr.open_dataset required by ecgtoos.builder.Builder
+        with xr.open_dataset(file, chunks={}, decode_times=False) as ds:
```

Comment:
As discussed earlier, this is certainly not the most efficient way, but I see that the default parser wants to fill in info like time:units, which is probably not essential for our use-case. There is scope to optimize, identifying the core columns, but not essential for this particular PR, given the scope.

Reply:
@wrongkindofdoctor here is an example of how our catalog builder gets the metadata, just for reference. This can further be improved, to take in a file-name-template or DRS-template rather than assuming CMIP, which will be super useful from the workflow angle. Note to self, mostly. 

https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/v1.0.0/intakebuilder/getinfo.py
https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/8ca820472997cc5bf2d1f2d8558dc7b084d5eecf/intakebuilder/localcrawler.py#L8

Reply:
@aradhakrishnanGFDL I can import the CatalogBuilder code and use it in place of the ecgtools for the GFDL class in the next iteration if that is preferable.